### DISCUSSION TITLE
Whether it is necessary for vehicle adapters to implement speed control

### DISCUSSION THREAD

**Discussion by ericbin** [openTCS/opentcs#80] [Q&A] [answered]
The vehicle LoopAdapter adapter uses the VelocityController for controlling the acceleration and speed of the vehicle. In my implementation of the actual hardware control, it's necessary to consider whether to include speed control. The data sent to the vehicle should only include location information, and then the vehicle can autonomously control its speed. Alternatively, a strategy for controlling the vehicle's speed can also be adopted.

**Comment by swltr** [comment, marked as answer]
The loopback driver contains that `VelocityController` only for its simulation of the vehicle's movement. For a real vehicle, the vehicle should only be told the destination and maybe parameters like maximum speed. Fine-grained speed control and steering is in the vehicle's responsibility.
